.github/workflows/documentation_workflow.yml

```diff
@@ -1,33 +1,40 @@
 name: Documentation
 on: [push, pull_request]
-
 jobs:
   MakeDocs:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - name: Setup R
         uses: r-lib/actions/setup-r@v2-branch
         with:
           r-version: '4.1.0'
       - name: Create Documentation directory
         run: |
-          echo "Creating 'Documentation'"
+          echo "Creating documentation directory"
           mkdir -p documentation
       - name: Install knitr dependency
         run: |
           echo "Installing R package dependencies. Scripts might have additional dependencies installed."
           Rscript -e 'if (!require(knitr)) install.packages("knitr")'
       - name: Documentation compilation
         run: 'Rscript .github/scripts/doc_builder.r'
-      - name: Commit Documentation
+      - name: Setup Git
+        run: |
+          echo "Setting up Git"
+          git config --global user.name github-actions[bot]
```

Comment:
Maybe you could get username and email from an environment variable? `GITHUB_ACTOR` or a similar one